### PR TITLE
nmap XML file parsing support

### PR DESCRIPTION
**Pull request by patriknordlen**
This PR addresses issue #218.

It adds a parser for nmap XML files, and creates informational findings with basic details about identified services from them.

It also extends the Endpoint model to include port and fqdn in order to improve filtering and reporting possibilities in the future.

### CODE REVIEW

File: dojo/models.py [hunk 1]
```diff
@@ -438,14 +440,19 @@ def __unicode__(self):
...
         query = self.query
         fragment = self.fragment
 
+        if port:
+            netloc += ':%s' % port
+
         if netloc or (scheme and scheme in uses_netloc and url[:2] != '//'):
             if url and url[:1] != '/': url = '/' + url
-            if scheme:
+            if scheme and scheme in uses_netloc and url[:2] != '//':
```
Comment (patriknordlen) [inline dojo/models.py]: I've done some testing against local files on this and it seems to work but the logic here is complicated enough that I don't fully trust my own judgment here...so please test this :)

**Comment by devGregA** [comment]
@patriknordlen will test first thing tomorrow

**Comment by devGregA** [comment]
@patriknordlen  tested and verified even using options such as -A. @patriknordlen would you please capitalize the 'n' in nmap in dojo/templates/dojo/import_scan_results.html, dojo/forms.py, and dojo/tools/factory.py for consistency  :) oh! and please add a line to https://github.com/OWASP/django-DefectDojo/blob/master/docs/features.rst#imports adding nmap

**Comment by patriknordlen** [comment]
Done @devGregA!

**Pull request merged**